zerver/models.py

```diff
@@ -1738,18 +1758,40 @@ class CustomProfileFieldValue(models.Model):
     class Meta(object):
         unique_together = ('user_profile', 'field')
 
+# A Service corresponds to either an outgoing webhook bot or an embedded bot.
+# The type of Service is specified by the interface field.
+#
+# If the Service is an outgoing webhook bot:
+# - name is any human-readable identifier for the Service
+# - base_url is the address of the third-party site
+# - token is used for authentication with the third-party site
+#
+# If the Service is an embedded bot:
+# - name is the canonical name for the type of bot (e.g. 'xkcd' for an instance
+#   of the xkcd bot); multiple embedded bots can have the same name, but all
+#   embedded bots with the same name will run the same code
+# - base_url is the address of the Zulip site
+# - token is the API key for the bot user corresponding to the Service
+#
+# N.B. After refactoring the bot library code to return the bot's reply (instead
+# of using a Client to send the reply), Services for embedded bots should no
+# longer need to store a base_url or a token.
 class Service(models.Model):
+    OUTGOING_WEBHOOK = 1
+    EMBEDDED = 2
+
     name = models.CharField(max_length=UserProfile.MAX_NAME_LENGTH) # type: Text
-    # owner of service/bot user corresponding to the service
+
+    # Bot user corresponding to the Service.  If non-bot services are added,
+    # user_profile can also represent the owner of the Service.
     user_profile = models.ForeignKey(UserProfile) # type: UserProfile
-    # address of the third party site
     base_url = models.TextField() # type: Text
-    # used for authentication to third party site
     token = models.TextField() # type: Text
-    # the interface used to send data to third party site
-    interface = models.PositiveSmallIntegerField(default=1)  # type: int
 
-    # Valid interfaces are {}
+    # Valid interfaces are:
+    # - OUTGOING_WEBHOOK
+    # - EMBEDDED
+    interface = models.PositiveSmallIntegerField(default=OUTGOING_WEBHOOK)  # type: int
```

Comment:
Interface here was for a separate thing initially. `Generic` class in [this](https://github.com/zulip/zulip/pull/3667/commits/f06c2448fc12377a700c352dedabfa0aef2582a4) commit is an example of interface. I don't know the current plan though. 
Also services can also be differentiated if they belong to embedded bot or outgoing webhook bot by `service.user_profile.is_embedded_bot`. If this differentiation is not required often, this method can be used. :)